### PR TITLE
For #11670: Upgrade Robolectric to 4.7.3

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by mcarare** [comment]
See also https://github.com/mozilla-mobile/android-components/issues/11816.

File: components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoResultTest.kt [hunk 1]
```diff
@@ -16,9 +16,12 @@ import org.junit.Assert.fail
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.mozilla.geckoview.GeckoResult
+import org.robolectric.annotation.LooperMode
 
+@Suppress("DEPRECATION")
 @ExperimentalCoroutinesApi
 @RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.LEGACY)
```
Comment (Amejia481) [inline components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoResultTest.kt:24]: Could we clarify what exactly we are suppressing the deprecation on ? I will be good to just add the deprecation on the specific item and link a ticket to letter address the issue :)

File: components/browser/engine-system/src/test/java/mozilla/components/browser/engine/system/SystemEngineSessionTest.kt [hunk 2]
```diff
@@ -45,10 +45,14 @@ import org.mockito.Mockito.never
...
 import org.mockito.Mockito.times
 import org.mockito.Mockito.verify
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.annotation.LooperMode
 import java.lang.reflect.Modifier
 import org.mockito.ArgumentMatchers.any as mockitoAny
 
+@Suppress("DEPRECATION")
 @RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.LEGACY)
```
Comment (Amejia481) [inline components/browser/engine-system/src/test/java/mozilla/components/browser/engine/system/SystemEngineSessionTest.kt:55]: Same as https://github.com/mozilla-mobile/android-components/pull/11801/files#r828032874

**Review by Amejia481 — APPROVED** [review]
Thanks for putting this patch 👏🏽 🏅  .
This looks good in overall, only nit to improve it's https://github.com/mozilla-mobile/android-components/pull/11801/files#r828032874

File: components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoResultTest.kt (hunk 1 above)
Comment (mcarare) [reply components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoResultTest.kt]: I added a comment to clarify the deprecation suppression is for the LEGACY LooperMode.

**Comment by mcarare** [comment]
I am going to keep this open for an additional day so that anyone could get the chance to add feedback if desired. 
After that, I will merge it in order to start work on https://github.com/mozilla-mobile/android-components/issues/9684 (removing `robolectric.properties` files, check tests, etc)

**Pull request merged**